### PR TITLE
Update namespaces for MEP converts and align to oM changes

### LINKED ISSUE #897: Update Namespaces for MEP converts

### PR DESCRIPTION
**Pull request by michaelhoehn**
### NOTE: Depends on 

https://github.com/BHoM/BHoM/pull/1069
https://github.com/BHoM/BHoM_Engine/pull/2133
   
### Issues addressed by this PR

Closes #897 

### Test files

[Here](https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/Revit_Toolkit/Revit_Toolkit-Issue897-MEP%20Converts%20Update%20Namespaces?csf=1&web=1&e=HrqIEn)

### Changelog

- Removing MEP.Node dependency within the convert code as this object class has been removed. Current solution requires `BH.oM.Geometry.Point`
- Sync all namespace changes
- Fix Wire convert

### CODE REVIEW

**Comment by michaelhoehn** [comment]
@BHoMBot check code compliance

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Review by pawelbaran — DISMISSED** [review]
That will conflict badly with #880 - how will we proceed with that @michaelhoehn @tiagogrossi @FraserGreenroyd?

**Comment by FraserGreenroyd** [comment]
> That will conflict badly with #880 - how will we proceed with that @michaelhoehn @tiagogrossi @FraserGreenroyd?

Whichever one is ready to merge first gets priority.

**Review by pawelbaran — COMMENTED** [review]
Following @tiagogrossi's thumbs up reaction, happy to unblock.

**Comment by michaelhoehn** [comment]
@pawelbaran thanks for the review. @tiagogrossi agreed in our last meeting that his cableTray PR will be updated after these changes, so this conflict should be expected.

File: Revit_Core_Engine/Convert/Environment/FromRevit/Pipe.cs [hunk 1]
```diff
@@ -41,25 +41,25 @@ public static partial class Convert
...
                 return bhomPipe;
 
             // Start and end points
             LocationCurve locationCurve = revitPipe.Location as LocationCurve;
             Curve curve = locationCurve.Curve;
-            BH.oM.MEP.Elements.Node startNode = new BH.oM.MEP.Elements.Node { Position = curve.GetEndPoint(0).PointFromRevit() }; // Start point
-            BH.oM.MEP.Elements.Node endNode = new BH.oM.MEP.Elements.Node { Position = curve.GetEndPoint(1).PointFromRevit() }; // End point
-            BH.oM.Geometry.Line line = BH.Engine.Geometry.Create.Line(startNode.Position, endNode.Position); // BHoM line
+            BH.oM.Geometry.Point startPoint = new BH.oM.Geometry.Point { X = curve.GetEndPoint(0).X, Y = curve.GetEndPoint(0).Y, Z = curve.GetEndPoint(0).Z };
+            BH.oM.Geometry.Point endPoint = new BH.oM.Geometry.Point { X = curve.GetEndPoint(1).X, Y = curve.GetEndPoint(1).Y, Z = curve.GetEndPoint(1).Z };
```
Comment (pawelbaran) [inline Revit_Core_Engine/Convert/Environment/FromRevit/Pipe.cs:57]: Shouldn't there be a call to `PointFromRevit()` to enforce unit conversion?

```suggestion
            BH.oM.Geometry.Point startPoint = curve.GetEndPoint(0).PointFromRevit();
            BH.oM.Geometry.Point endPoint = curve.GetEndPoint(1).PointFromRevit();
```

File: Revit_Core_Engine/Convert/Environment/FromRevit/Wire.cs [hunk 2]
```diff
@@ -37,33 +37,28 @@ public static partial class Convert
...
             if (bhomWire != null)
                 return bhomWire;
             
             LocationCurve locationCurve = revitWire.Location as LocationCurve;
             Curve curve = locationCurve.Curve;
-            BH.oM.MEP.Elements.Node startNode = new BH.oM.MEP.Elements.Node { Position = curve.GetEndPoint(0).PointFromRevit() }; // Start point
-            BH.oM.MEP.Elements.Node endNode = new BH.oM.MEP.Elements.Node { Position = curve.GetEndPoint(1).PointFromRevit() }; // End point
-            BH.oM.Geometry.Line line = BH.Engine.Geometry.Create.Line(startNode.Position, endNode.Position); // BHoM line
+            BH.oM.Geometry.Point startPoint = new BH.oM.Geometry.Point { X = curve.GetEndPoint(0).X, Y = curve.GetEndPoint(0).Y, Z = curve.GetEndPoint(0).Z };
+            BH.oM.Geometry.Point endPoint = new BH.oM.Geometry.Point { X = curve.GetEndPoint(1).X, Y = curve.GetEndPoint(1).Y, Z = curve.GetEndPoint(1).Z };
```
Comment (pawelbaran) [inline Revit_Core_Engine/Convert/Environment/FromRevit/Wire.cs:57]: Shouldn't there be a call to `PointFromRevit()` to enforce unit conversion?

```suggestion
            BH.oM.Geometry.Point startPoint = curve.GetEndPoint(0).PointFromRevit();
            BH.oM.Geometry.Point endPoint = curve.GetEndPoint(1).PointFromRevit();
```

File: Revit_Core_Engine/Convert/Environment/FromRevit/Wire.cs [hunk 3]
```diff
@@ -37,33 +37,28 @@ public static partial class Convert
...
+            BH.oM.Geometry.Line line = BH.Engine.Geometry.Create.Line(startPoint, endPoint); // BHoM line
 
             // Wire
-            bhomWire = new BH.oM.MEP.Elements.Wire();
-
-            // Wire segment
-            WireSegment wireSegment = BH.Engine.MEP.Create.Wire(line);
-
-            bhomWire.WireSegments.Add(wireSegment);
+            bhomWire = BH.Engine.MEP.Create.WireSegment(line);
```
Comment (pawelbaran) [inline Revit_Core_Engine/Convert/Environment/FromRevit/Wire.cs:61]: Some code disappeared here, is that on purpose?

**Review by pawelbaran — CHANGES_REQUESTED** [review]
The name of the PR is confusing - it is not a mere namespace change, it also introduces functional changes. Besides that, I've spotted a few minor issues, please see the code comments.

File: Revit_Core_Engine/Convert/Environment/FromRevit/Wire.cs (hunk 3 above)
Comment (michaelhoehn) [reply Revit_Core_Engine/Convert/Environment/FromRevit/Wire.cs:61]: @pawelbaran, yes this is intentional here. Wires were left unfinished from the initial convert creation and did not function as expected, so a minor code adjustment provides missing functionality here. Also, hoping this change helps clear up the bit of ambiguity between the Wire and WireSegment objects.

**Comment by michaelhoehn** [comment]
@pawelbaran, updated PR title and documentation to reflect more accurately, as well as the nice adjustment to point creation. LMK if this is now more clear or if more description needs to be added. 😄

**Review by pawelbaran — APPROVED** [review]
The code LGTM now, but I have not run any tests - I will leave the second, functional review to the MEP guys 👍

**Review by kayleighhoude — APPROVED** [review]
Happy to approve from a pipe/duct perspective, everything's pulling across properly.
image

**Review by FraserGreenroyd — APPROVED** [review]
LGTM

**Pull request merged**